broker/src/main/java/io/camunda/zeebe/broker/partitioning/Partition.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.partitioning;
+
+import io.atomix.raft.partition.RaftPartition;
+import io.camunda.zeebe.broker.partitioning.startup.PartitionStartupContext;
+import io.camunda.zeebe.broker.partitioning.startup.steps.PartitionDirectoryStep;
+import io.camunda.zeebe.broker.partitioning.startup.steps.RaftBootstrapStep;
+import io.camunda.zeebe.broker.partitioning.startup.steps.SnapshotStoreStep;
+import io.camunda.zeebe.broker.partitioning.startup.steps.ZeebePartitionStep;
+import io.camunda.zeebe.broker.system.partitions.ZeebePartition;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.scheduler.startup.StartupProcess;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents the combination of a Raft and Zeebe partition. For now, the only way to construct
+ * partitions is through the {@link #bootstrapping(PartitionStartupContext)} method.
+ *
+ * <pre>
+ *   final var context = new PartitionStartupContext(...);
+ *   final var partition = Partition.bootstrapping(context);
+ *
+ *   partition.start().join();
+ *   partition.stop().join();
+ * </pre>
+ */
+public final class Partition {
+  private static final Logger LOGGER = LoggerFactory.getLogger(Partition.class);
+
+  private final PartitionStartupContext context;
+  private final StartupProcess<PartitionStartupContext> startupProcess;
+
+  private Partition(
+      final PartitionStartupContext context,
+      final StartupProcess<PartitionStartupContext> startupProcess) {
+    this.context = context;
+    this.startupProcess = startupProcess;
+  }
+
+  /**
+   * Creates a partition that uses the bootstrapping process when started.
```

Comment:
:wrench: May be explain what you mean by bootstrapping process. I think what you mean here is : bootstrapping = when the partition is already part of the replication group. Right?